basetest.pm

```diff
@@ -325,15 +328,26 @@ sub run_post_fail {
         # Read them now to not stumble upon them in the next module.
         $self->get_new_serial_output();
     }
+
     $self->fail_if_running();
+    $self->compute_test_execution_time();
+    my $post_fail_hook_execution_time = execution_time($post_fail_hook_start_time);
+    bmwqemu::diag(sprintf("||| post fail hooks runtime: %d s", $post_fail_hook_execution_time));
     die $msg . "\n";
 }
 
 sub execution_time { time - shift }
 
+sub compute_test_execution_time {
+    my ($self) = @_;
```

Comment:
It would be nice to use signatures in new code.

Reply:
@Martchus I will try to add signatures on a separate commit, as no methods have them within this package, let's see how it goes.

Reply:
We should be good to go then 🏎️